daemon/cmd/daemon.go

```diff
@@ -353,6 +353,8 @@ func NewDaemon(ctx context.Context, epMgr *endpointmanager.EndpointManager, dp d
 		d.redirectPolicyManager,
 	)
 
+	d.redirectPolicyManager.RegisterSvcCache(&d.k8sWatcher.K8sSvcCache)
```

Comment:
Minor: you can pass svcCache as an argument in NewRedirectPolicyManager at L 344 above. Also, since we are storing it in the manager, we might as well use it at other places in the manager.

Reply:
at L344, the watcher has not been created, hence no svc cache available (created alongside with watcher) yet.

Reply:
Ahok! We can still use the stored svcCache at other places, though?

Reply:
I think so, redirect manager has the handle to it, so should be straightforward to expand the interface if needed.

Reply:
:) We could accommodate those changes in this PR itself. It can help with unit testing down the road. That's up to you.

Reply:
Done.

Reply:
@Weil0ng @aditighag If we added a 5s sleep above this line, would it trigger any odd behaviour because the `d.redirectPolicyManager.svcCache` is not yet initialized?

Reply:
I don't think so? So prior to L 356, both redirect policy manager and K8s watcher are instantiated. But the K8s watcher subsystem is initialized (thereby all the watchers and handlers) at a later point - https://github.com/cilium/cilium/blob/master/daemon/cmd/daemon.go#L569

However, we need to add more test cases around this code. It's in the future improvements list. For example, I would like to verify that the redirect policy manager event handlers are invoked at right places.